### DISCUSSION TITLE
Redux => MobX migration case study.

### DISCUSSION THREAD

**Discussion by jtwigg** [mobxjs/mobx#3479] [Show and tell]
I just finished migrating from a Redux store to a Mobx store and thought I'd post some of my results here.

The chief reason I started the migration was 
A) Redux boiler plate was out of control: Adding a simple "set boolean" to my state would require modifying multiple files and creating multiple declarations (Action ID's and `action` functions, reducers, etc)
B) React re-draws were out of control: Despite reading all the docs, I could never get redux re-draw's under control.

In total, the delta in code was
```
MobX: 790 lines of code
Redux: 1600 lines of code
```

So roughly 1:2 for MobX:Redux

However: This doesn't capture the other benefits
* My current MobX implementation actually does more. Redraws are correct and appropriate. I now have better async/thunk like behavior than I did previously. 
* Everything is much easier to reason about.

**So thank you all for helping me get my app state under control!**

**Comment by pervezalam777** [comment]
I really wish to see what you have done, if it is possible for you to share some gist of your code.

**Comment by kubk** [comment]
Thank you so much for the write up! 👍
